zeebe/engine/src/main/java/io/camunda/zeebe/engine/processing/user/IdentitySetupInitializer.java

```diff
@@ -31,26 +27,16 @@
 import org.springframework.security.crypto.password.PasswordEncoder;
 
 public final class IdentitySetupInitializer implements StreamProcessorLifecycleAware, Task {
-  public static final String DEFAULT_USER_USERNAME = "demo";
-  public static final String DEFAULT_USER_PASSWORD = "demo";
-  public static final String DEFAULT_USER_EMAIL = "demo@demo.com";
   public static final String DEFAULT_ROLE_NAME = "Admin";
   public static final String DEFAULT_TENANT_ID = TenantOwned.DEFAULT_TENANT_IDENTIFIER;
   public static final String DEFAULT_TENANT_NAME = "Default";
   private static final Logger LOG = Loggers.PROCESS_PROCESSOR_LOGGER;
   private final SecurityConfiguration securityConfig;
   private final PasswordEncoder passwordEncoder;
-  private final RoleState roleState;
-  private final UserState userState;
-  private final TenantState tenantState;
 
-  public IdentitySetupInitializer(
-      final SecurityConfiguration securityConfig, final MutableProcessingState processingState) {
+  public IdentitySetupInitializer(final SecurityConfiguration securityConfig) {
     this.securityConfig = securityConfig;
     passwordEncoder = PasswordEncoderFactories.createDelegatingPasswordEncoder();
-    userState = processingState.getUserState();
-    roleState = processingState.getRoleState();
-    tenantState = processingState.getTenantState();
   }
 
   @Override
```

Comment:
@remcowesterhoud , is this condition still valid?
```
    if (!securityConfig.getAuthorizations().isEnabled()) {
      // If authorization is disabled we don't need to setup identity.
      LOG.debug("Skipping identity setup as authorization is disabled");
      return;
    }
```

how can I call the API if I don't have some initial credentials to pass the authentication checks?

Reply:
when I use `auth-basic` profile, authentication checks are [enabled](https://github.com/camunda/camunda/blob/c4a7a53e628d8c0b13ff3e94570ed983db738062/authentication/src/main/java/io/camunda/authentication/config/WebSecurityConfig.java#L100-L138), any calls to authenticated paths like `/v2/**` should be authenticated.

When I use an unauthenticated client I get:
```
DEBUG org.springframework.web.servlet.mvc.method.annotation.HttpEntityMethodProcessor - Writing [{timestamp=Wed Dec 11 16:08:56 CET 2024, status=401, error=Unauthorized, path=/v2/users}]
```

Reply:
@houssain-barouni I'm confused what your point is here? With the `auth-basic` profile authorization checks are enabled. That makes sense because you say you want to use basic auth. Since it enables authentications a default user is created. If you use an unauthenticated client of course you'll be unauthorized. You just enabled basic auth so you need to add that to the client?

Reply:
@remcowesterhoud yes, there is no correlation between `auth-basic` spring profile and authorizations
`auth-basic` just means that the users are created an managed inside Camunda, and not by a third-party OIDC or LDAP.
we can use `auth-basic` without having authorizations enabled. auth stands for authentication.

Reply:
So there is no way to run Camunda without authentication anymore? That seems not very user friendly?

Reply:
I think we may have another profile reserved for that @Ben-Sheppard .
Anyway, I think we can have a property `camunda.security.initialization.defaultuser.enabled` set to `false` by default and set to `true` for `auth-basic`. I think for the oidc case we don t need to create a default user (maybe you need to create default mapping roles...)

Reply:
> Anyway, I think we can have a property `camunda.security.initialization.defaultuser.enabled` set to false

The `defaultUser` property is gone. It's now just a list of users. If no users are configured it will create a default demo/demo one.

Reply:
>The defaultUser property is gone. It's now just a list of users. If no users are configured it will create a default demo/demo one.

We need another condition then to disable the data initialization